pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -712,6 +733,10 @@ func (nc *Controller) monitorNodeHealth() error {
 			continue
 		}
 
+		nc.nodeHealthMap[node.Name].observedReadyCondition = &observedReadyCondition
```

Comment:
I think it's quite weird that you need to store both observedReadyCondition and currentReadyCondition.

Please also not that tryUpdateNodeHealth is already updating nc.nodeHealthMap so maybe the data you need is already there or you can extend that place to keep more data if needed.

Reply:
I've removed the additional fields. It turns out:
- grace period can have only one value - we use it only when currentReadyCondition != nil.
- observedReadyCondition is not needed. IMHO we can use only currentReadyCondition here.
tryUpdateNodeHealth method will update node statues in case currentReadyCondition is different than observedReadyCondition, else (if currentReadyCondition !=nil - our case) observedReadyCondition is a copy of currentReadyCondition.
- currentReadyCondition is calculated based on saved node status.